src/plugins/intel_npu/src/compiler/include/graph_transformations.hpp

```diff
@@ -14,9 +17,39 @@
  */
 namespace intel_npu::driverCompilerAdapter {
 
-/**
- * @brief Serialize OpenVINO model to IR
- */
-IR serializeToIR(const std::shared_ptr<const ov::Model>& model, uint32_t supportedVersionByCompiler = 7);
+class IRSerializer {
```

Comment:
Cosmetic: the code probably would be clearer if we pass the model directly instead of stashing it in the helper object, so would consider replacing this class with several free functions for readability

Reply:
The reason to save model here is because we may clone the model, then call counter and writer may clone the model twice. Otherwise, have to make the condition check and model clone to zero_compiler_in_driver.cpp and make the serializeIR function harder to read